src/core.js

```diff
@@ -850,6 +850,34 @@ export default function Core(rootElement, userSettings, rootInstanceSymbol = fal
     }
   }
 
+  /**
+   * Internal function to set `className` or `tableClassName` key of settings.
+   *
+   * @private
+   * @param {Object} settings
+   */
+  function setClassName(settings) {
+    if (settings.className) {
+      if (GridSettings.prototype.className) {
+        removeClass(instance.rootElement, GridSettings.prototype.className);
+      }
+
+      addClass(instance.rootElement, settings.className);
+
+      GridSettings.prototype.className = settings.className;
+    }
+
+    if (settings.tableClassName && instance.table) {
+      if (GridSettings.prototype.tableClassName) {
+        removeClass(instance.table, GridSettings.prototype.tableClassName);
+      }
+
+      addClass(instance.table, settings.tableClassName);
+
+      GridSettings.prototype.tableClassName = settings.tableClassName;
+    }
+  }
```

Comment:
In the worse scenario, you change class names 4 times. It might be decreased to max 2 reflows (one addClass and one removeClass).

BTW. I don't like passing `settings` object in that way. We need only two properties, so why not to pass only them?
BTW2 We can set these class names inside `updateSettings` as I mentioned in other comments.

Reply:
We add and remove the class in two different elements. That is 4 times is necessary sometimes, I suppose.
`settings` I need in inner body of the function and I don't have access to them for outside, so I decided to pass in the parameter.
But after your comment, I have tried to get and slightly change `setClassName` function to more compact.